File: AÑO 2/programacion_y_diseño_orientado_a_objetos/Practicas/Napakalaki_Ruby/lib/bad_consequence.rb
#encoding: utf-8

class BadConsequence
  attr_reader :text, :levels
  
  @@MAXTREASURES = 10
  def self.MAXTREASURES
    @@MAXTREASURES
  end
  
  def initialize(text,levels)
    @text=text
    @levels=levels
  end
  
  def to_s
    "BadConsequence: text=#{@text}, levels=#{@levels}";
  end
end

File: AÑO 2/programacion_y_diseño_orientado_a_objetos/Practicas/Napakalaki_Java/src/NapakalakiGame/BadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.ArrayList;

/**
 *
 * @author guillermogotre
 */
public abstract class BadConsequence {
    final static int MAXTREASURES = 10;
    private String text;
    private int levels;     
    
    public BadConsequence(String text, int levels){
        this.text = text;
        this.levels = levels;
    }
    
    public abstract boolean isEmpty();
    
    public String getText() {
        return text;
    }

    public int getLevels() {
        return levels;
    }
    
    public abstract void substractVisibleTreasure(Treasure t);
    
    public abstract void substractHiddenTreasure(Treasure t);
    
    public abstract BadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v, ArrayList<Treasure> h);

    @Override
    public String toString() {
        return text + "\nlevels: " + levels;
    }
    
    
}
